File: Python/OOP/Abstraction/Abstraction (example 1).py
# Python ≥3.4 is required to run this program
from abc import ABC, abstractmethod
class Student(ABC):
    @abstractmethod
    def studentInfo():
        pass

    def academicInfo(self):
        print("Dept. of Computer")

class Naiem(Student):
    def studentInfo(self):
        print("Name: Abdullah Al Naiem")

class abstraction:
    obj = Naiem()
    obj.studentInfo()
    obj.academicInfo()

File: Java/OOP/Abstraction/Abstraction (example 1).java
abstract class Student {
    abstract void studentInfo();
    void academicInfo(){
        System.out.println("Dept. of Computer");
    }
}
class Naiem extends Student {
    @Override // annotation
    public void studentInfo(){
        System.out.println("Name: Abdullah Al Naiem");
    }
}
class abstraction {
    public static void main(String[] args){
        Student obj = new Naiem();
        obj.studentInfo();
        obj.academicInfo();
    }
}